src/helm/proxy/clients/anthropic_client.py

```diff
@@ -18,13 +19,145 @@
 from .client import Client, wrap_request_time, truncate_sequence
 
 
+class AnthropicClient(Client):
+    """
+    Client for the Anthropic models (https://arxiv.org/abs/2204.05862).
+    They use their own version of the GPT-2 tokenizer.
+    """
+
+    MAX_COMPLETION_LENGTH: int = 8192
+    ADDITIONAL_TOKENS: int = 5
+    PROMPT_ANSWER_START: str = "The answer is"
+
+    def __init__(self, api_key: str, cache_config: CacheConfig, tokenizer_client: Client):
+        self.api_key: str = api_key
+        self.cache = Cache(cache_config)
+        self.tokenizer_client: Client = tokenizer_client
+        self._client = anthropic.Client(api_key)
+
+    def _send_request(self, raw_request: Dict[str, Any]) -> Dict[str, Any]:
+        result = self._client.completion(**raw_request)
+        assert "error" not in result, f"Request failed with error: {result['error']}"
+        return result
+
+    def _filter_completion(self, completion: str, max_tokens: int) -> str:
+        # If the completion starts with a colon, space, or newline, remove it.
+        for _ in range(AnthropicClient.ADDITIONAL_TOKENS):
+            if completion[0] in [":", " ", "\n"]:
+                completion = completion[1:]
+            else:
+                break
+
+        # NOTE(josselin): For now, this is disabled as it is not an accurate
+        # limit of tokens. It is still handled by truncate_sequence() further
+        # down the line, but it is not ideal. (warnings are disabled)
```

Comment:
I've elaborated more. However I am not satistfied with one of the changes made. Since we are now changing the number of tokens in the run expander I feel like the completion is not properly truncated anymore. The idea was:
- You want to query at least $x$ tokens.
- Query $x + \delta$ tokens.
- Remove up to the first $\delta$ tokens if they are colons, spaces or newlines.
- Truncate to $x$ tokens.

With the run expander, I feel like we are now missing the last step.
This would again require some kind of "post-processing" run expander.